slack_sdk/socket_mode/aiohttp/__init__.py

```diff
@@ -0,0 +1,193 @@
+        auto_reconnect_enabled: bool = True,
+        ping_interval: float = 10,
+        on_message_listeners: Optional[List[Callable[[WSMessage], None]]] = None,
+        on_error_listeners: Optional[List[Callable[[WSMessage], None]]] = None,
+        on_close_listeners: Optional[List[Callable[[WSMessage], None]]] = None,
+    ):
+        self.app_token = app_token
+        self.logger = logger or logging.getLogger(__name__)
+        self.web_client = web_client or AsyncWebClient()
+        self.proxy = proxy
+        self.auto_reconnect_enabled = auto_reconnect_enabled
+        self.ping_interval = ping_interval
+
+        self.wss_uri = None
+        self.web_socket_message_queue = Queue()
+        self.web_socket_message_listeners = []
+        self.socket_mode_request_listeners = []
+        self.current_session = None
+        self.current_session_monitor = None
+
+        self.on_message_listeners = on_message_listeners or []
+        self.on_error_listeners = on_error_listeners or []
+        self.on_close_listeners = on_close_listeners or []
+
+        self.web_socket_message_receiver = None
+        self.web_socket_message_processor = asyncio.ensure_future(
+            self.process_web_socket_messages()
+        )
+
+    async def monitor_current_session(self) -> None:
+        while True:
+            await asyncio.sleep(self.ping_interval)
+            try:
+                if self.auto_reconnect_enabled and (
+                    self.current_session is None or self.current_session.closed
+                ):
+                    self.logger.info(
+                        "The session seems to be already closed. Going to reconnect..."
+                    )
+                    await self.connect_to_new_endpoint()
+            except Exception as e:
+                self.logger.error(
+                    "Failed to check the current session or reconnect to the server "
+                    f"(error: {type(e).__name__}, message: {e})"
+                )
+
+    async def receive_web_socket_messages(self) -> None:
+        consecutive_error_count = 0
+        while True:
+            try:
+                message: WSMessage = await self.current_session.receive()
+                if self.logger.level <= logging.DEBUG:
+                    type = WSMsgType(message.type)
+                    message_type = type.name if type is not None else message.type
+                    message_data = message.data
+                    if isinstance(message_data, bytes):
+                        message_data = message_data.decode("utf-8")
+                    self.logger.debug(
+                        f"Received message (type: {message_type}, data: {message_data}, extra: {message.extra})"
+                    )
+                if message is not None:
+                    if message.type == WSMsgType.TEXT:
+                        message_data = message.data
+                        await self.enqueue_web_socket_message(message_data)
```

Comment:
the main responsibility of this method is to enqueue received messages into the internal queue.